File: frontend/src/api/reviews.ts
import { apiFetch } from './client'

export type RoutineName = 'morning' | 'nightly' | 'weekly'

export interface RoutineReview {
  review_id: string
  correlation_id: string
  routine: RoutineName
  target_date: string
  routine_status: 'published_claude' | 'published_fallback' | 'late_upgraded'
  provider: 'claude_subscription' | 'deterministic'
  review_text: string
  structured: Record<string, unknown>
  action_ids: string[]
  partial_actions: Array<Record<string, unknown>>
  published_at: string
  claude_session_url?: string
}

export function fetchReviews(limit = 20) {
  return apiFetch<{ reviews: RoutineReview[] }>(`/api/reviews?limit=${limit}`)
}

export function fetchReview(routine: RoutineName, targetDate: string) {
  return apiFetch<{ review: RoutineReview }>(
    `/api/reviews/${routine}/${targetDate}`,
  )
}

File: core/hub/reviews.py
"""Browser-safe shaping of published routine reviews.

A stored review carries internal fields the Hub has no business receiving, so
the client view is an explicit allowlist rather than a blocklist: a field added
to the store is invisible to the browser until someone deliberately lists it.
"""
from __future__ import annotations

import logging

logger = logging.getLogger(__name__)


_REVIEW_CLIENT_FIELDS = frozenset({
    "review_id",
    "correlation_id",
    "routine",
    "target_date",
    "routine_status",
    "provider",
    "review_text",
    "structured",
    "action_ids",
    "partial_actions",
    "published_at",
})


def _review_for_client(review: dict, runs) -> dict:
    """Return a browser-safe canonical review with optional Claude navigation."""
    from core.review_delivery import normalise_claude_session_url

    result = {
        key: value for key, value in review.items() if key in _REVIEW_CLIENT_FIELDS
    }
    session_url = normalise_claude_session_url(review.get("claude_session_url"))
    if not session_url and review.get("correlation_id"):
        run = runs.get(str(review["correlation_id"])) or {}
        session_url = normalise_claude_session_url(run.get("claude_session_url"))
        if not session_url:
            remote = run.get("remote_trigger_result")
            session_url = normalise_claude_session_url(
                remote.get("claude_code_session_url")
                if isinstance(remote, dict)
                else None
            )
    if session_url:
        result["claude_session_url"] = session_url
    return result
